programs/ziti-edge-tunnel/instance.c

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2019 Netfoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "model/dtos.h"
+#include "ziti/sys/queue.h"
+
+struct tnl_identity_s {
+    tunnel_identity *id;
+    LIST_ENTRY(tnl_identity_s) _next;
+};
+
+static LIST_HEAD(tnl_identities, tnl_identity_s) tnl_identity_list = LIST_HEAD_INITIALIZER(&tnl_identity_list);
+
+tunnel_identity get_tunnel_identity(ziti_identity *identity) {
```

Comment:
should return pointer to allocated struct

Reply:
I have changed it to pointer in the next PR - https://github.com/openziti/ziti-tunnel-sdk-c/pull/200/files